_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,329 @@
+
+Let's create our `topics.json` file in a file in the current directory of our local machine:
+```json
+{
+  "version": 1,
+  "topics": [
+    { "topic": "my-topic-two"}
+  ]
+}
+```
+
+After creating this file, copy it to the interactive pod container since we will be running all of our commands from there:
+
+```sh
+kubectl cp topics.json my-pod:/tmp/topics.json
+```
+
+We can now start a shell process in our interactive pod container and run the command inside it to generate our proposal `reassignment.json` data.
+Let's start the shell process:
+
+```sh
+kubectl exec -ti my-pod /bin/bash
+```
+
+Now it's time to generate the `reassignment.json` file. 
+We will use the `kafka-reassign-partitions.sh` command to generate our proposal `reassignment.json` data:
+
+```sh
+kubectl exec -ti my-pod /bin/bash -- bin/kafka-reassign-partitions.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 \
+--topics-to-move-json-file /tmp/topics.json \
+--broker-list 0,1,2,3 \
+--generate
+```
+Here `topics-to-move-json-file` points towards the `topics.json file` and `--broker-list` is the list of brokers we want to move our partitions onto.
+
+Once you run this command, you will be able to see the JSON data which is generated by the Kafka reassignment partition tool. You get the current replica assignment and the proposed `reassignment.json` data.
+
+```sh
+Current partition replica assignment
+{"version":1,"partitions":[{"topic":"my-topic-two","partition":0,"replicas":[3,4,2,0],"log_dirs":["any","any","any","any"]},{"topic":"my-topic-two","partition":1,"replicas":[0,2,3,1],"log_dirs":["any","any","any","any"]},{"topic":"my-topic-two","partition":2,"replicas":[1,3,0,4],"log_dirs":["any","any","any","any"]}]}
+
+Proposed partition reassignment configuration
+{"version":1,"partitions":[{"topic":"my-topic-two","partition":0,"replicas":[0,1,2,3],"log_dirs":["any","any","any","any"]},{"topic":"my-topic-two","partition":1,"replicas":[1,2,3,4],"log_dirs":["any","any","any","any"]},{"topic":"my-topic-two","partition":2,"replicas":[2,3,4,0],"log_dirs":["any","any","any","any"]}]}
+
+```
+
+You can copy this proposed partition reassignment and paste it in a local `reassignment.json` file
+
+Edit the `reassignment.json` to remove a replica from each partition
+
+```sh
+echo "$( jq 'del(.partitions[].replicas[3])' reassignment.json)" > reassignment.json
+```
+
+We then copy the `reassignment.json` file to the interactive pod container.
+
+```sh
+kubectl cp reassignment.json my-pod:/tmp/reassignment.json
+```
+Now we'll start a shell process inside the interactive pod container to run our Kafka bin script.
+
+```sh
+kubectl exec -ti my-pod /bin/bash -- bin/kafka-reassign-partitions.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 \
+--reassignment-json-file /tmp/reassignment.json \
+--execute
+```
+Because removing replicas from a broker doesn't require any interbroker data movement there is no need to supply a `--throttle` in this case.
```

Comment:
```suggestion
Because removing replicas from a broker doesn't require any interbroker data movement there is no need to supply a `--throttle` in this case.
(Interbroker data movement would be needed if you were increasing the replication factor, so in that case a `--throttle` would be recommended).
```